File: app/src/main/java/DictionaryBuilder/DictionaryPage.java
package DictionaryBuilder;

import java.io.File;
import java.io.IOException;

public abstract class DictionaryPage {
    private File outputFile;

    public DictionaryPage(String outputPath){
        this.outputFile = new File(outputPath);
        this.outputFile.getParentFile().mkdirs();
    }

    public File getOutFile(){
        return outputFile;
    }

    public abstract void build() throws IOException;
}

File: app/src/main/java/dictionary/DictionaryPage.kt
package dictionary

import java.io.File
import java.io.IOException

abstract class DictionaryPage(outputPath: String) {
    val outFile: File

    init {
        outFile = File(outputPath)
        outFile.parentFile.mkdirs()
    }

    @Throws(IOException::class)
    abstract fun build()
}